backend/src/web_server.py

```diff
@@ -57,7 +57,8 @@ def start(self):
             self.socket.bind((self.hostname, self.port))
 
         except socket.error as err:
-            print("ERROR! binding port {self.port} failed with error: {err}")
+            print(f"ERROR! binding port {self.port} failed with error: {err}")
+            self.socket.close()
```

Comment:
I think shutdown before close is the order of operations.

Reply:
Yes you are correct! Do you want me to add it to the shutdown function instead so it happens every time shutdown is called?